### ISSUE TITLE
Multiple pinned notes can be pushed to building flag position

### ISSUE THREAD

**Issue by tothxa** [widelands/widelands#5788]
**Describe the bug**
Pinned notes are not allowed on buildings, the note is moved to the building's flag instead. However if the flag already has a note, then both notes will share the same position. If repeated multiple times, then all notes will be placed to the same spot. It also happens if the 2 flags are placed before the building is built. When there are multiple flags at the same position, only the last one can be edited. As deleting it also works, they can be removed one by one.

**To reproduce**

a.
1. Place a pinned note on a building's flag (to make it better visible, pick a colour and use a short text)
2. Place a pinned note on the building itself (with different colour and longer text for visibility)
3. See error

b.
1. Place pinned notes at a free building spot and its flag position (make them different for better visibility of the problem)
2. Place a building on the spot
3. See error

**Expected behavior**
For case a., the existing note on the flag should be edited when trying to place a new one.
I'm not sure about case b. The simplest solution would be to remove the note at the building position, if this is acceptable.

Alternatively, just allow notes on buildings? (can't remember whether disallowing them was a technical requirement or just a design decision)

**Comment by Noordfrees** [comment]
The reason why notes mustn't live on buildings is that they are inaccessible there since clicking on a building will open the buildingwindow and not the fieldactionwindow. We should never autodelete a note, players might use them to pin important information to the map and this should not be lost unless the player explicitly deletes the note. 

Another option could be that when we click on a field with a note on it, we directly open the note editor, and not show a fieldactionwindow or buildingwindow. Then we can allow notes to exist on buildings too.

**Comment by tothxa** [comment]
> The reason why notes mustn't live on buildings is that they are inaccessible there since clicking on a building will open the buildingwindow and not the fieldactionwindow.

Well, we have the `F8` fastplace-like button for them, so they are accessible with that. I always use that anyway. It just needs to be added to the fieldaction window icon's tooltip. (which should already have been done anyway :)

> Another option could be that when we click on a field with a note on it, we directly open the note editor, and not show a fieldactionwindow or buildingwindow. Then we can allow notes to exist on buildings too.

Not being able to access buildings and field actions is worse than pinned notes isn't it? :)

Couldn't the watch tab code from field actions be moved to a new class, so that building windows would have a watch tab too? I think it makes sense for buildings too anyway.

**Comment by tothxa** [comment]
And now that ships will use them, pinned notes should get a window too, where all of them are listed, with edit and go to actions. That would also solve the accessibility problem.
